index.js

```diff
@@ -171,28 +171,47 @@ module.exports = {
   handlers(self, options) {
     return {
       beforeSave: {
-        async beforeSaveHandler(req, piece, options) {
+        async denormalizeDateTimes(req, piece, options) {
           self.denormalizeDatesAndTimes(piece)
         }
       },
+      beforeInsert: {
+        setGroupId(req, piece, options) {
+          // Set groupId on parent if this is a repeating item
+          if (piece.dateType === 'repeat') {
+            piece.groupId = self.apos.util.generateId()
+          }
+        }
+      },
       afterInsert: {
-        async afterInsertHandler(req, piece, options) {
+        async createRepeatItems(req, piece, options) {
           if (self._workflowPropagating) {
-            // if (self.isClone) {
             // Workflow is replicating this but also its existing
             // scheduled repetitions, don't re-replicate them and cause problems
-            // Don't allow replication of an event that is already a clone
-            console.log('Already replicating')
             return
           }
-          if (piece.dateType === 'repeat') {
+          if (piece.dateType === 'repeat' && piece.aposMode === 'draft') {
             await self.repeatEvent(req, piece, options)
             return
           } else {
-            console.log('Non repeating event')
             return
           }
         }
+      },
+      afterPublish: {
+        async publishChildren(req, piece, options) {
+          // If this is a repeating item, publish its children also
+          console.log('dataType', piece.dataType)
+          if (piece.dataType === 'repeat') {
+            console.log('Publishing', piece)
+            const existing = await self.findChildren(req, {
```

Comment:
I see you are not actually trying to publish these yet. But note that you will get an infinite loop if you don't exclude the original piece when doing so — it will also have the groupId so there is a risk of getting it over and over. You can use a `_id: { $ne: piece._id } clause in your criteria to exclude it from the results for "children."

Reply:
Also you should check `options.firstTime`, if it is not set you should not autopublish the children — after the main event is published for the first time, anyone editing it probably does not expect their updates to result in the repeated events getting published too. Later that will change if we introduce a way to intentionally duplicate the changes to the repeated events.